Add `clip-shadow-above` configuration and wintype option

tryone144 opened the pull request:
Added the new `shadow-crop-window` configuration option and `crop-shadow` wintype option. These allow the user to select windows to crop from the shadow region of other windows, i.e. don't paint shadows on top of them.

This should provide a more useful and user friendly alternative to the deprecated `shadow-exclude-reg` option.
See https://github.com/yshui/picom/issues/494 and https://github.com/yshui/picom/pull/582#issuecomment-809261483.

---

The name of this new option is up for debate. My rationale behind not calling it `shadow-exclude-window` was to avoid confusion with the similarly named `shadow-exclude` option that disables the shadow that a window casts. In contrast, the new option prevents shadows from being cast onto the target window.
I am not a native speaker, so if *crop* does not intuitively convey they right meaning, *clip* might work as well (or *trim*, *truncate*, …).

**EDIT:** Renamed the options to `clip-shadow-above`.

yshui commented:
Thanks for implementing this!

As for the name, I am not sure either. I am thinking about something like `above-shadow`, meaning the window will be above shadows, what do you think? Otherwise I think `clip-shadow` is probably better.

yshui commented:
@tryone144 Yeah, that sounds good :+1:

tryone144 commented:
Renamed to `clip-shadow-above` and rebased.
Merging this should provide a solution to https://github.com/yshui/picom/issues/494 and (possibly) https://github.com/yshui/picom/issues/4.

yshui commented:
Thank you!

joaopauloalbq commented:
Any possibility to clip shadow above shadow and thus avoid overlapping shadows?
